flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/InternalExecutionGraphAccessor.java

```diff
@@ -122,4 +122,6 @@ List<ShuffleDescriptor> getClusterPartitionShuffleDescriptors(
             IntermediateDataSetID intermediateResultPartition);
 
     MarkPartitionFinishedStrategy getMarkPartitionFinishedStrategy();
+
+    boolean isHybridEnableConsumePartialFinishedProducer();
```

Comment:
I'm not entirely sure about introducing a hybrid specific interface in execution graph. Haven't think of an alternative way though.

Reply:
Yes, I also didn't find a better way to handle this. Let's keep it first, If there is a better way, we can reconstruct it later.